### PR TITLE
Deserialize trajectory connectivity

### PR DESCRIPTION
**Pull request by ojura**

### CODE REVIEW

File: cartographer/mapping_3d/sparse_pose_graph.h [hunk 1]
```diff
@@ -188,8 +188,7 @@ class SparsePoseGraph : public mapping::SparsePoseGraph {
   void LogResidualHistograms() REQUIRES(mutex_);
 
   // Updates the trajectory connectivity structure with the new constraints.
```
Comment (wohe) [inline cartographer/mapping_3d/sparse_pose_graph.h]: Please update the comment. Same in 2D.

File: cartographer/mapping_2d/sparse_pose_graph.cc [hunk 2]
```diff
@@ -321,7 +318,9 @@ void SparsePoseGraph::HandleWorkQueue() {
         RunOptimization();
 
         common::MutexLocker locker(&mutex_);
-        UpdateTrajectoryConnectivity(result);
+        for (const auto& constraint : result) {
```
Comment (wohe) [inline cartographer/mapping_2d/sparse_pose_graph.cc]: No need for `auto` here: `Constraint`?

**Comment by ojura** [comment]
Updated.

**Comment by wohe** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user wohe. Merge queue now has a length of 1.

**Pull request merged**